src/visitors/MethodAndRemoteMethodCountVisitor.java

```diff
@@ -0,0 +1,40 @@
+package visitors;
+
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.MethodDeclaration;
+import com.github.javaparser.ast.expr.MethodCallExpr;
+import com.github.javaparser.ast.visitor.GenericVisitorAdapter;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class MethodAndRemoteMethodCountVisitor extends GenericVisitorAdapter<Integer, Void> {
+
+    @Override
+    public Integer visit(ClassOrInterfaceDeclaration n, Void arg) {
+        // returns number of methods and methodCallExprs within those methods
+        return n.getMethods().size() + super.visit(n, arg);
```

Comment:
`super.visit(n, arg)` atm probably is returning just the no. of maximum remote calls any method has instead of accumulating.
If you are adding the remote calls in each method, you need to make sure that each unique remote call is counted only once, I believe.

Reply:
Also there is a case where a method is calling another method in the same class. Would your code count it?

Reply:
All the method call expressions are stored in a **set**, so they only counted once.